src/plugins/home/server/services/sample_data/lib/sample_dataset_schema.ts

```diff
@@ -92,6 +92,12 @@ export const sampleDataSchema = schema.object({
 
   status: schema.maybe(schema.string()),
   statusMsg: schema.maybe(schema.string()),
+
+  // index settings do not work in serverless
+  withIndexSettings: offeringBasedSchema({
+    traditional: schema.boolean({ defaultValue: true }),
+    serverless: schema.boolean({ defaultValue: false }),
+  }),
```

Comment:
For that kind of ES capabilities related switches, you could eventually use the [ES capabilities feature](https://github.com/elastic/kibana/pull/164850) instead